assets/js/gov-overrides.js

```diff
@@ -0,0 +1,17 @@
+//Overrides GOVUK exitPage function so it open in newtab
+if (typeof ExitThisPage === 'function') { //check ExitThisPage Class Exists
+  ExitThisPage.prototype.exitPage = function() {
+
+    //Loading Overlay copied from govuk
+    document.body.classList.add("govuk-exit-this-page-hide-content");
+    this.$overlay = document.createElement("div");
+    this.$overlay.className = "govuk-exit-this-page-overlay";
+    this.$overlay.setAttribute("role", "alert");
+    document.body.appendChild(this.$overlay);
+    this.$overlay.textContent = this.i18n.t("activated");
+
+    //Functionality to Open in new Tab
+    window.open("http://bbc.co.uk/weather", "_newtab");
+    window.location.replace("http://www.google.co.uk");  
+  };
+}
```

Comment:
Can we also add in a function override for the shift keys to get them to behave in the same way?  

The current GDS function is 
```
  ExitThisPage.prototype.handleKeypress = function (event) {
    // Detect if the 'Shift' key has been pressed. We want to only do things if it
    // was pressed by itself and not in a combination with another key—so we keep
    // track of whether the preceding keyup had shiftKey: true on it, and if it
    // did, we ignore the next Shift keyup event.
    //
    // This works because using Shift as a modifier key (e.g. pressing Shift + A)
    // will fire TWO keyup events, one for A (with e.shiftKey: true) and the other
    // for Shift (with e.shiftKey: false).
    if (
      (event.key === 'Shift' || event.keyCode === 16 || event.which === 16) &&
      !this.lastKeyWasModified
    ) {
      this.keypressCounter += 1

      // Update the indicator before the below if statement can reset it back to 0
      this.updateIndicator()

      // Clear the timeout for the keypress timeout message clearing itself
      if (this.timeoutMessageId !== null) {
        clearTimeout(this.timeoutMessageId)
        this.timeoutMessageId = null
      }

      if (this.keypressCounter >= 3) {
        this.keypressCounter = 0

        if (this.keypressTimeoutId !== null) {
          clearTimeout(this.keypressTimeoutId)
          this.keypressTimeoutId = null
        }

        this.exitPage()
      } else {
        if (this.keypressCounter === 1) {
          this.$updateSpan.innerText = this.i18n.t('pressTwoMoreTimes')
        } else {
          this.$updateSpan.innerText = this.i18n.t('pressOneMoreTime')
        }
      }

      this.setKeypressTimer()
    } else if (this.keypressTimeoutId !== null) {
      // If the user pressed any key other than 'Shift', after having pressed
      // 'Shift' and activating the timer, stop and reset the timer.
      this.resetKeypressTimer()
    }

    // Keep track of whether the Shift modifier key was held during this keypress
    this.lastKeyWasModified = event.shiftKey
  }

}
```

Reply:
This override already works for the shift keys aswell. if you see in the function it calls this.exitPage()

Reply:
It doesn't seem to work - at least not for me locally.  I looked at the code and expected it to work but it didn't.  Is it working for you?  
(Quick Exit worked, just didn't open the other tab.)

Reply:
It did for me. Are you sure your popup blocker isnt stopping it..It did for me on local the first time.